src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -640,6 +640,115 @@ vtkSmartPointer< vtkDataSet > manageGlobalIds( vtkSmartPointer< vtkDataSet > mes
   return output;
 }
 
+/**
+ * @brief This function tries to make sure that no MPI rank is empty
+ *
+ * @param[in] mesh a vtk grid
+ * @param[in] comm the MPI communicator
+ * @return the vtk grid redistributed
+ */
+vtkSmartPointer< vtkDataSet >
+ensureNoEmptyRank( vtkDataSet & mesh,
+                   MPI_Comm const comm )
+{
+  // step 1: figure out who is a donor and who is a recipient
+  globalIndex const numElems = mesh.GetNumberOfCells();
+  globalIndex const numProcs = MpiWrapper::commSize( comm );
```

Comment:
I think we prefer to use `int` or `integer` to represent rank/partition numbers (they don't need to be 64-bit), here and below. Also `numElems` and related quantities should probably be `localIndex`, since we're describing local mesh parts.
```suggestion
  localIndex const numElems = LvArray::integerConverstion< localIndex >( mesh.GetNumberOfCells() );
  int const numProcs = MpiWrapper::commSize( comm );
```

Reply:
Done, I updated the types in the new function